app/src/main/java/org/simple/clinic/summary/PatientSummaryScreenController.kt

```diff
@@ -125,15 +125,61 @@ class PatientSummaryScreenController @Inject constructor(
         .map { (_, patientUuid) -> { ui: Ui -> ui.showUpdatePrescribedDrugsScreen(patientUuid) } }
   }
 
-  private fun handleBackClicks(events: Observable<UiEvent>): Observable<UiChange> {
-    return events
+  private fun backAndDoneClicks(events: Observable<UiEvent>): Observable<UiChange> {
+    val patientUuids = events
+        .ofType<PatientSummaryScreenCreated>()
+        .map { it.patientUuid }
+
+    val bloodPressureSaves = events
+        .ofType<PatientSummaryBloodPressureSaved>()
+        .startWith(PatientSummaryBloodPressureSaved(false))
```

Comment:
Hrm, `PatientSummaryBloodPressureSaved` sounds like an event that is emitted when a BP is saved. Can we rename this to `PatientSummaryBloodPressureClosed` or something?

Reply:
```PatientSummaryBloodPressureSaved``` is an event that is emitted only when Blood pressure is saved.